packages/labs/analyzer/src/lib/model.ts

```diff
@@ -304,33 +403,64 @@ export type ClassHeritage = {
 export interface ClassDeclarationInit extends DeclarationInit {
   node: ts.ClassDeclaration;
   getHeritage: () => ClassHeritage;
+  fieldMap?: Map<string, ClassField> | undefined;
+  methodMap?: Map<string, ClassMethod> | undefined;
 }
 
 export class ClassDeclaration extends Declaration {
   readonly node: ts.ClassDeclaration;
   private _getHeritage: () => ClassHeritage;
   private _heritage: ClassHeritage | undefined = undefined;
+  readonly _fieldMap: Map<string, ClassField> | undefined;
+  readonly _methodMap: Map<string, ClassMethod> | undefined;
 
   constructor(init: ClassDeclarationInit) {
     super(init);
     this.node = init.node;
     this._getHeritage = init.getHeritage;
+    this._fieldMap = init.fieldMap;
+    this._methodMap = init.methodMap;
   }
 
   get heritage(): ClassHeritage {
     return (this._heritage ??= this._getHeritage());
   }
+
+  get fields() {
+    return Array.from(this._fieldMap?.values() ?? []);
+  }
+
+  get methods() {
+    return Array.from(this._methodMap?.values() ?? []);
+  }
+
+  getField(name: string): ClassField | undefined {
+    return this._fieldMap?.get(name);
+  }
+
+  getMethod(name: string): ClassMethod | undefined {
+    return this._methodMap?.get(name);
+  }
+
+  getMember(name: string): ClassMethod | ClassField | undefined {
+    return this.getField(name) ?? this.getMethod(name);
```

Comment:
This actually implements a little bit of prototype lookup in that fields are own properties of the instance and methods are on the prototype. So you could have a field and method of the same name and get correct resolution here.

ie, this is valid JS:

```js
class A {
  x = 1;
  x() {}
}
```

Worth a comment?

Reply:
Done.

In TS that's an error, but it's possible when analyzing JS I guess. Added comment describing same-named fields as taking precedence due to being instance properties.